File: Org Of Programming Languages/week 8/HW2 problem 3.pl
divisiable(DD,DV) :- DV > 1, 0 =:= (DD mod DV).
divisiable(DD,DV) :- DV-1 > 1, divisiable(DD,DV-1).

isPrime(2) :- true,!.
isPrime(X) :- X < 2, false,!.
isPrime(X) :- X >= 3, not(divisiable(X,X // 2)).

File: Org Of Programming Languages/week 8/HW2 problem 3.cpp
#include <stdio.h>

bool isPrime(int);

int main() {
	int num;
	char tmp;
	bool done = false;
	while (!done) {
		printf("\nEnter a number to test to see if it's prime or 0 to exit:");
		scanf("%d", &num);
		scanf("%c", &tmp);
		if (num > 0) {
			if (isPrime(num)) {
				printf("The number is prime!!\n");
			}
			else {
				printf("The number is not prime\n");
			}
		}
		else if (num == 0) {
			done = true;
		}
	}
	return 0;
}

bool isPrime(int num) {
	if (num == 1 || num == 2 || num == 3)
		return true;

	for (int i = (int) num / 2; i > 1; i--) {
		if (num % i == 0) {
			return false;
		}
	}

	return true;
}